cmd/ndm_daemonset/probe/deleteDeviceHandler.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"github.com/openebs/node-disk-manager/blockdevice"
+	apis "github.com/openebs/node-disk-manager/pkg/apis/openebs/v1alpha1"
+	"k8s.io/klog"
+)
+
+func (pe *ProbeEvent) deleteBlockDevice(bd blockdevice.BlockDevice, bdAPIList *apis.BlockDeviceList) error {
+	pe.deleteDeviceFromCache(bd)
+
+	// legacy UUID will be generated at event processing and filled into blockdevice UUID field, no need to regerate here.
+	klog.V(4).Infof("device: %s disconnected, generated legacy UUID: %s.", bd.DevPath, bd.UUID)
+
+	uuid, ok := generateUUID(bd)
+	// this can happen if the device didn't have any unique identifiers
+	if !ok {
+		klog.Info("could not recreate GPT-UUID while removing device")
+	}
+
+	// try to deactivate resource with both UUIDs. This is required in the following case
```

Comment:
why we are trying both the method, gpt and legacy? We should try one of them right?

Reply:
No, Consider the following case where the device using legacy UUID has been unclaimed. We will create resource with new UUID. So if it is disconnected we need to deactivate both of them. This is a kind of fail-safe mechanism that is used.

Ideally, after unclaiming blockdevice that used legacy UUID, user should restart the NDM pod , so that everything comes to sync.